### PR TITLE
Refactor our use of Vault

### PR DESCRIPTION
**Pull request by KellerFuchs**
- [x] Avoid needlessly opening Vault-encrypted files.
  This should avoid having Ansible start super-slowly and prompt for the GPG passphrase even when no secrets are needed.
- [x] Only make secrets available when needed, and be clearer on where the data is coming from.
- [x] Avoid relying on hardcoded lists of encrypted files in `.gitattributes` and `vault_passphrase.sh`.
  We have previously ran into trouble when encrypted files were added to the repository but not the lists.
- [x] Use consistent conventions for file locations and variable names.
- [x] Consistently provide an unencrypted skeleton of the file, so the structure of the data is available in plain text.

I don't know if this is the best way to do it, so feel free to comment.
In particular, I would like to know if the structure is easier to understand, and whether the explanations in `vault/README.md` are any good.

### CODE REVIEW

**Comment by KellerFuchs** [comment]
~The CI failure seems spurious, but I will look when I'm more available.~

Confirmed: this issue was that `ansible-lint` ignores all `-x` options except the last one.
I fixed our build script.

**Review by daurnimator — APPROVED** [review]

**Review by DeviaVir — APPROVED** [review]
:+1:

**Comment by KellerFuchs** [comment]
@daurnimator @DeviaVir Do the approvals mean it's more readable now?

**Comment by DeviaVir** [comment]
@KellerFuchs yes, all of these changes make it better imo.

**Comment by daurnimator** [comment]
It's no worse than it was before; and I don't see any issues with it. Go for it :)

**Comment by KellerFuchs** [comment]
Mmmh, Travis seems to be borked  :(
Edit: Restarted a build, seems to have done the trick.

File: lint.sh [hunk 1]
```diff
@@ -3,25 +3,33 @@
...
-    [shell.yml]="-x ANSIBLE0006"
-    [ldap_ban.yml]="-x ANSIBLE0012"
+# Per-file lint ignores
+declare -A F_IGNORES=(
+    [shell.yml]="ANSIBLE0006"
+    [ldap_ban.yml]="ANSIBLE0012"
 )
 
-exec() {
+function exec() {
```
Comment (RyanSquared) [inline lint.sh:15]: This one still has parenthesis where the `function join` below does not &lt;/nit&gt;

Comment (KellerFuchs) [reply lint.sh:15]: Indeed!
Send a PR?  :D

**Pull request merged**